crates/orchestrator/src/jobs/snos_job/fact_topology.rs

```diff
@@ -23,34 +23,45 @@ pub struct FactTopology {
 
 /// Returns the fact topology from the additional data of the output builtin.
 pub fn get_fact_topology(cairo_pie: &CairoPie, output_size: usize) -> Result<FactTopology, FactError> {
+    tracing::debug!("Starting get_fact_topology function");
     if let Some(BuiltinAdditionalData::Output(additional_data)) = cairo_pie.additional_data.0.get(&BuiltinName::output)
     {
+        tracing::trace!("Found Output additional data");
         let tree_structure = match additional_data.attributes.get(GPS_FACT_TOPOLOGY) {
             Some(tree_structure) => {
+                tracing::debug!("GPS_FACT_TOPOLOGY found in additional data attributes");
                 ensure!(!tree_structure.is_empty(), FactError::TreeStructureEmpty);
                 ensure!(tree_structure.len() % 2 == 0, FactError::TreeStructureLenOdd);
                 ensure!(tree_structure.len() <= 10, FactError::TreeStructureTooLarge);
                 ensure!(tree_structure.iter().all(|&x| x < 2 << 30), FactError::TreeStructureInvalid);
+                tracing::trace!("Tree structure validation passed");
                 tree_structure.clone()
             }
             None => {
+                tracing::warn!("GPS_FACT_TOPOLOGY not found in additional data attributes");
                 ensure!(additional_data.pages.is_empty(), FactError::OutputPagesLenUnexpected);
+                tracing::debug!("Using default tree structure");
                 vec![1, 0]
             }
         };
+        tracing::debug!("Retrieving page sizes");
         let page_sizes = get_page_sizes(&additional_data.pages, output_size)?;
+        tracing::info!("FactTopology successfully created");
```

Comment:
either this needs to be a debug or in the starting of the function we need a info, also if using info, follow the pattern
```suggestion
        tracing::debug("FactTopology successfully created");
```

Reply:
valid.